#! /bin/sh -e
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Corrects pkg-config information 

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0
@DPATCH@

diff -urN valgrind-3.3.0.orig/valgrind.pc.in valgrind-3.3.0/valgrind.pc.in
--- valgrind-3.3.0.orig/valgrind.pc.in	2007-12-10 23:18:47.000000000 +0000
+++ valgrind-3.3.0/valgrind.pc.in	2007-12-12 20:38:54.000000000 +0000
@@ -4,13 +4,13 @@
 includedir=@includedir@/valgrind
 arch=@VG_ARCH@
 os=@VG_OS@
-platform=@VG_PLATFORM_PRI@
+platform=@VG_ARCH@-linux
 valt_load_address=@VALT_LOAD_ADDRESS@
 
 Name: Valgrind
 Description: A dynamic binary instrumentation framework
 Version: @VERSION@
 Requires:
-Libs: -L${libdir}/valgrind/@VG_PLATFORM_PRI@ -lcoregrind -lvex -lgcc
+Libs: -L${libdir}/valgrind/@VG_ARCH@-linux -lcoregrind -lvex -lgcc
 Cflags: -I${includedir}
 
